lib/compress.js

```diff
@@ -706,9 +706,9 @@ merge(Compressor.prototype, {
         return x;
     };
 
-    function is_iife_call(node) {
+    function is_iife_call(node, traditional) {
         if (node instanceof AST_Call && !(node instanceof AST_New)) {
-            return is_func_expr(node.expression) || is_iife_call(node.expression);
+            return is_func_expr(node.expression, traditional) || is_iife_call(node.expression);
```

Comment:
Thinking about this some more - are you trying to suppress `negate_iife` to emit `!` upon `async`? Because this looks like it's suppressing `AST_Arrow` in general instead.

And if so, might as well just use `node.expression instanceof AST_Function` instead of `is_func_expr()` here.